sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobQueryReader.java

```diff
@@ -74,6 +77,32 @@ public Flux<ByteBuffer> read() {
             .concatMap(this::parseRecord);
     }
 
+    /**
+     * Avro parses a query reactive stream.
+     *
+     * The Avro stream is formatted as the Avro Header (that specifies the schema) and the Avro Body (that contains
+     * a series of blocks of data). The Query Avro schema indicates that the objects being emitted from the parser can
+     * either be a result data record, an end record, a progress record or an error record.
+     *
+     * @return The parsed query reactive stream.
+     */
+    public InputStream readInputStream(InputStream inputStream) throws IOException {
+        // Convert InputStream to Flux<ByteBuffer>
+        Flux<ByteBuffer> avroFlux = toFluxByteBuffer(inputStream);
+
+        // Use existing read method to process the data
+        Flux<ByteBuffer> processedData = new AvroReaderFactory().getAvroReader(avroFlux).read()
```

Comment:
Further out, should track adding synchronous APIs to Avro parsing. At a high-level this method feels odd as it is converting an `InputStream` to `Flux<ByteBuffer>` to process data to convert back to an `InputStream`.

Reply:
Will follow up with this in the PR when converting internal-avro.